File: my-solutions/0013-roman-to-integer/solution.py
class Solution:
    def romanToInt(self, s):
        result = 0
        i = 0
        n = len(s)
        
        while i < n:
            s1 = self.value(s[i])
            
            if i + 1 < n:
                s2 = self.value(s[i + 1])
                if s1 >= s2:
                    result += s1
                else:
                    result += s2 - s1
                    i += 1
            else:
                result += s1
                
            i += 1
            
        return result
    
    def value(self, r):
        roman_values = {
            'I': 1,
            'V': 5,
            'X': 10,
            'L': 50,
            'C': 100,
            'D': 500,
            'M': 1000
        }
        return roman_values.get(r, -1)

File: my-solutions/0013-roman-to-integer/solution.java
class Solution {
    public int romanToInt(String s) {
        int result = 0;
        for (int i = 0; i < s.length(); i++) {
            int s1 = value(s.charAt(i));

            if (i + 1 < s.length()) {
                int s2 = value(s.charAt(i + 1));

                if (s1 >= s2) {
                    result += s1;
                } else {
                    result += s2 - s1;
                    i++; // Пропускаем следующую цифру, так как мы её уже обработали
                }
            } else {
                result += s1;
            }
        }
        return result;
    }

    static int value(char r) {
        return switch (r) {
            case 'I' -> 1;
            case 'V' -> 5;
            case 'X' -> 10;
            case 'L' -> 50;
            case 'C' -> 100;
            case 'D' -> 500;
            case 'M' -> 1000;
            default -> -1;
        };
        }

        
    }


